lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -54,88 +54,3 @@ function ( $item ) {
 		}
 	}
 }
-
-/**
- * Returns the stylesheet resulting of merging core, theme, and user data.
- *
- * @param array $types Types of styles to load. Optional.
- *                     It accepts 'variables', 'styles', 'presets' as values.
- *                     If empty, it'll load all for themes with theme.json support
- *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
- *
- * @return string Stylesheet.
- */
-function gutenberg_get_global_stylesheet( $types = array() ) {
-	// Return cached value if it can be used and exists.
-	// It's cached by theme to make sure that theme switching clears the cache.
-	$can_use_cached = (
```

Comment:
This logic is changing:

- :heavy_check_mark: adds a filter.
- :heavy_check_mark: keeps `empty( $types )`.
- :heavy_check_mark: keeps the `WP_DEBUG` check.
- :heavy_check_mark: removes the `SCRIPT_DEBUG`. `WP_DEBUG` should be enough.
- :heavy_check_mark: removes `is_admin`. I don't know why we started adding `is_admin` in a few places (in this function, [in the SVG filters cache](https://github.com/WordPress/gutenberg/blob/c9df6b412c37a968fb23bd99402b5b485c69d8a3/lib/compat/wordpress-6.0/get-global-styles-and-settings.php#L79), and [webfonts](https://github.com/WordPress/gutenberg/blob/c9df6b412c37a968fb23bd99402b5b485c69d8a3/lib/experimental/register-webfonts-from-theme-json.php#L16)). From the cache management point of view, the way it works now is that admin users won't use the cache, no matter the other conditions. I don't see a reason why, other than perhaps circumventing the issues introduced by the transient (styles were not immediately applied in the front-end). I think it's safe to remove.
- :yellow_circle: removes `REST_REQUEST`. I don't know why. After some searching, I found [the PR](https://github.com/WordPress/gutenberg/pull/25680/files) that first introduced this logic. The SVG and webfonts PRs came later, so they probably followed what they saw. I asked there for more context.

Reply:
Got a response from Ari: he says the rest request check was probably for the editor (it was [many moons](https://github.com/WordPress/gutenberg/pull/25680/) ago, so it's hard for anyone to remember exactly). Going to investigate this a bit.

Reply:
> @geriux and @fluiddot would you be able to test if this PR, as it is, works for you as well? If so, we can safely remove it.

Sure! I will test it and report back 👍